### PR TITLE
Pass compiler_flags to GHC during linking of Haskell binaries

### PR DESCRIPTION
**Pull request by gdeest**
As discussed in #91 , some compiler flags such as `-threaded` MUST be passed during the linking phase. This PR solves this problem by passing `compiler_flags` to GHC during linking.

Do we want to add a test for this change ?

### CODE REVIEW

**Comment by mboes** [comment]
Can you include a test please?

**Comment by gdeest** [comment]
Sure thing.

File: tests/BUILD [hunk 1]
```diff
@@ -40,6 +40,13 @@ rule_test(
   size = "small",
 )
 
+rule_test(
```
Comment (mboes) [inline tests/BUILD]: This rule is redundant. It doesn't test any property that shouldn't already be covered by other tests. It's the `sh_test` below that tests a useful property.

Comment (gdeest) [reply tests/BUILD]: Granted.

**Pull request merged**